src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/CodeActions/Razor/GenerateMethodCodeActionProvider.cs

```diff
@@ -116,4 +137,49 @@ private static bool IsGenerateEventHandlerValid(
         methodName = content;
         return true;
     }
+
+    private static bool TryGetEventNameAndMethodName(
+        MarkupTagHelperAttributeSyntax markupTagHelperDirectiveAttribute,
+        TagHelperBinding binding,
+        [NotNullWhen(true)] out string? methodName,
+        [NotNullWhen(true)] out string? eventName,
+        out string? eventParameterType)
+    {
+        methodName = null;
+        eventName = null;
+        eventParameterType = null;
+
+        foreach (var tagHelperDescriptor in binding.Descriptors)
+        {
+            foreach (var attribute in tagHelperDescriptor.BoundAttributes)
+            {
+                if (attribute.Name == markupTagHelperDirectiveAttribute.TagHelperAttributeInfo.Name)
+                {
+                    if (!attribute.IsEventCallbackProperty())
+                    {
+                        return false;
+                    }
+
+                    // TypeName is something like "EventCallback<System.String>", so we need to parse out the parameter type.
+                    if (ComponentAttributeIntermediateNode.TryGetEventCallbackArgument(attribute.TypeName.AsMemory(), out var argument))
```

Comment:
(not related to this pr)

me: Why is this memory?

also me: ohh... why does that function take memory instead of a span?

Reply:
Compiler team: Chasing perf at all costs!
Tooling team: `.ToString()` all the things, and then serialize them to Json just in case!